### PR TITLE
npm1300 mfd: reset and hibernate

### PR DESCRIPTION
**Pull request by aasinclair**
Added reset and hibernate for npm1300

### CODE REVIEW

File: drivers/reset/reset_npm1300.c [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#define NPM_MAIN_BASE 0x00U
+
+/* nPM1300 GPIO registers offsets */
+#define NPM_MAIN_OFFSET_SWRESET 0x01U
+
+struct reset_npm1300_config {
+	const struct device *mfd;
+};
+
+static int reset_npm1300_toggle(const struct device *dev, uint32_t id)
```
Comment (gmarull) [inline drivers/reset/reset_npm1300.c:24]: `ARG_UNUSED(id)`?

I wonder if the reset API is well suited for this device (not sure what reset does in this case). The Reset API is oriented to IP blocks that control the reset lines of certain peripherals. See for example the DT-based API, where you'll typically have reset clients via `resets = <&resetctl ID>`. You can find example usages in ST/GD32 drivers.

Comment (aasinclair) [reply drivers/reset/reset_npm1300.c:24]: In this case it forces a full reset of the PMIC.  I agree it is tricky to decide where to put it.

Comment (gmarull) [reply drivers/reset/reset_npm1300.c:24]: maybe as an API in the mfd driver?

Comment (aasinclair) [reply drivers/reset/reset_npm1300.c:24]: How about adding reset & hibernate to the mfd driver (as they are fairly specific to the npm1300), and adding ship to regulators (as it is common to many devices)?

Comment (gmarull) [reply drivers/reset/reset_npm1300.c:24]: sounds good to me

**Review by gmarull — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**